wsclient/src/main/java/org/hpccsystems/ws/client/platform/Version.java

```diff
@@ -1,15 +1,36 @@
+/*******************************************************************************
+ * HPCC SYSTEMS software Copyright (C) 2020 HPCC Systems®.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
 package org.hpccsystems.ws.client.platform;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class Version implements Comparable<Version>
 {
-    public String versionString = "";
-    public String prefix        = "";
-    public int    major         = 0;
-    public int    minor         = 0;
-    public int    point         = 0;
-    public String postfix       = "";
-    public String postfixStr    = "";
-    public int    postfixInt    = 0;
+    private String versionString = "";
+    private String project        = "";
+    private int    major         = -1;
+    private int    minor         = -1;
+    private int    point         = -1;
+    private int    sequence      = -1;
+    private String maturity      = "";
+
+    final String regex = "(?:(?<project>[a-zA-Z-]*)_)?(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<point>\\d+)(?:\\-(?<presequence>\\d+)?(?:(?<maturity>(?i)rc)?(?<postsequence>\\d+)?))?";
```

Comment:
Can we provide an example of what we are matching above the regex pattern here? Otherwise regex itself looks good